pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/TableConfigsRestletResource.java

```diff
@@ -230,6 +235,11 @@ public ConfigSuccessResponse addConfig(
       try {
         _pinotHelixResourceManager.addSchema(schema, false, false);
         LOGGER.info("Added schema: {}", schema.getSchemaName());
+
+        if (typesToSkip == null) {
+          // Validate config again with schema for validations requires schema.
+          validateConfig(tableConfigs, databaseName, null);
```

Comment:
Do we need to revert the schema creation as validation failure will fail the request?

Reply:
yes, the catch clause will revert.

Reply:
Ah sorry missed that